package-res/resources/web/pentaho/type/property.js

```diff
@@ -0,0 +1,1331 @@
+           *
+           * @see pentaho.type.Complex#countRange
+           *
+           * @see pentaho.type.spec.IPropertyTypeProto#countMin
+           */
+          countMin: {
+            value: 0,
+            cast: castCount,
+            combine: function(baseEval, localEval) {
+              return function() {
+                return Math.max(baseEval.call(this), localEval.call(this));
+              };
+            }
+          },
+
+          /**
+           * Evaluates the value of the `countMax` attribute of a property of this type
+           * on a given complex value.
+           *
+           * This method is used to determine the effective
+           * [element count range]{@link pentaho.type.Property.Type#countRangeEval} and
+           * is not intended to be used directly.
+           *
+           * @name countMaxEval
+           * @memberOf pentaho.type.Property.Type#
+           * @param {pentaho.type.Complex} owner The complex value that owns a property of this type.
+           * @return {number} The evaluated value of the `countMax` attribute.
+           *
+           * @ignore
+           */
+
+          /**
+           * Gets or sets the maximum number of elements that properties of this type can have.
+           *
+           * A non-negative integer.
+           *
+           * Note that this attribute is taken together with
+           * the [isRequired]{@link pentaho.type.Property.Type#isRequired}
+           * and the [countMin]{@link pentaho.type.Property.Type#countMin} attributes
+           * to determine the effective [element count range]{@link pentaho.type.Property.Type#countRangeEval}
+           * of a _property type_.
+           *
+           * ### This attribute is *Dynamic*
+           *
+           * When a _dynamic_ attribute is set to a function,
+           * it can evaluate to a different value for each complex instance.
+           *
+           * When a _dynamic_ attribute is set to a value other than a function or a {@link Nully} value,
+           * its value is the same for every complex instance.
+           *
+           * ### This attribute is *Monotonic*
+           *
+           * The value of a _monotonic_ attribute can change, but only in some, predetermined _monotonic_ direction.
+           *
+           * In this case, a _property type_ having a `countMax` of `5` can later be
+           * changed to have a `countMax` of `3`.
+           * However, a _property type_ having a `countMax` of `5` can no longer "go back" to
+           * have a greater `countMax`, like `7`.
+           * The `countMax` attribute can only change to a lower value.
+           *
+           * Because this attribute is also _dynamic_,
+           * the actual `countMax` values are only known
+           * when evaluated for specific complex instances.
+           * As such, ensuring monotonic changes is deferred until evaluation.
```

Comment:
Suggest:

Because of this behavior, ensuring monotonic changes are deferred until evaluation.